boards: arm: Add Laird Connectivity BL5340 DVK platform

nordicjm opened the pull request:
This adds initial support for the Laird Connectivity BL5340
development kit hardware

Signed-off-by: Jamie McCrae <jamie.mccrae@lairdconnect.com>

ioannisg reviewed (approved):
This looks fine to me, please, take this out of draft and add the 2.6 milestone, if not there already

ioannisg commented:
@lairdjm you will have to look into the build failures, and also, I suppose this PR requires the TF-M module update. So I guess you can rebase this PR on top of #34774

ioannisg commented:
@lairdjm #34774 is updated, please, try to re-run this CI here

ioannisg commented:
> @lairdjm #34774 is updated, please, try to re-run this CI here

Nevermind @lairdjm , I did the rebase myself.

anangl commented on boards/arm/bl5340_dvk/bl5340_dvk_cpunet_reset.c:
```diff
@@ -0,0 +1,85 @@
+#define CPUNET_UARTE_PORT_RCTS NRF_P1
+#endif
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#include <../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h>
+#else
+#define DEBUG_SETUP()
+#endif
```
`BT_CTLR_DEBUG_PINS_CPUAPP` is an option only available for nRF5340 DK, see:
https://github.com/zephyrproject-rtos/zephyr/blob/0923e32fe743e71103ac660a50fd6fb2c5968ef6/subsys/bluetooth/controller/Kconfig#L625-L627
so I think this can be removed (together with the call to `DEBUG_SETUP()` in [line 50](https://github.com/zephyrproject-rtos/zephyr/pull/34833/files#diff-04678edb346d8cf445ddeb02e3394986d59514b695884e60c594187c48ac8eefR50)).

anangl reviewed (commented):
Looks good, only some minor issues spotted.

rerickson1 reviewed (changes requested):
commit with title `arch: arm: Add Laird Connectivity BL5340 DVK platform` should be changed to `boards: arm: Add Laird Connectivity BL5340 DVK platform`

ioannisg commented:
I saw a failure earlier on the adc test @lairdjm did you fix that?

ioannisg commented:
@lairdjm could you please rebase once more ?

Commit:
boards: arm: Add Laird Connectivity BL5340 DVK platform

This adds initial support for the Laird Connectivity BL5340
development kit hardware

Signed-off-by: Jamie McCrae <jamie.mccrae@lairdconnect.com>
